libfwupd/fwupd-client.c

```diff
@@ -428,7 +366,90 @@ fwupd_client_connect (FwupdClient *self, GCancellable *cancellable, GError **err
 	if (val != NULL)
 		fwupd_client_set_host_security_id (self, g_variant_get_string (val, NULL));
 
-	return TRUE;
+	/* success */
+	g_task_return_boolean (task, TRUE);
+}
+
+static void
+fwupd_client_connect_get_bus_cb (GObject *source,
+				 GAsyncResult *res,
+				 gpointer user_data)
+{
+	g_autoptr(GTask) task = G_TASK (user_data);
+	FwupdClient *self = g_task_get_source_object (task);
+	FwupdClientPrivate *priv = GET_PRIVATE (self);
+	g_autoptr(GError) error = NULL;
+
+	priv->conn = g_bus_get_finish (res, &error);
+	if (priv->conn == NULL) {
+		g_prefix_error (&error, "Failed to connect to system D-Bus: ");
+		g_task_return_error (task, g_steal_pointer (&error));
+		return;
+	}
+	g_dbus_proxy_new (priv->conn,
+			  G_DBUS_PROXY_FLAGS_NONE,
+			  NULL,
+			  FWUPD_DBUS_SERVICE,
+			  FWUPD_DBUS_PATH,
+			  FWUPD_DBUS_INTERFACE,
+			  g_task_get_cancellable (task),
+			  fwupd_client_connect_get_proxy_cb,
+			  g_object_ref (task));
+}
+
+/**
+ * fwupd_client_connect_async:
+ * @self: A #FwupdClient
+ * @cancellable: the #GCancellable, or %NULL
+ * @callback: the function to run on completion
+ * @callback_data: the data to pass to @callback
+ *
+ * Sets up the client ready for use. This is probably the first method you call
+ * when wanting to use libfwupd.
```

Comment:
"wanting to use libfwupd in an asynchronous manner"

Reply:
Fixed, thanks.